### DISCUSSION TITLE
could not compile `log`

### DISCUSSION THREAD

**Discussion by HipyCas** [SeaQL/sea-orm#692] [Q&A] [answered]
## Description

When trying to build a project that uses Rocket, I get a compilation error on the `log` crate, which must be caused by `sea-orm`, as without the `sea-orm` dependencies no error is reported. When hovering over the reported error on the Cargo.toml file, it says: _build failed error: could not compile `kv-log-macro`_

## Steps to Reproduce

I recommend to look and run the code in the repository that I've set up for it, as it really doesn't have specific steps to reproduce it. The repository is: https://github.com/HipyCas/rust-server-test

### Expected Behavior

Just work, as it does with the example.

### Actual Behavior

Simply doesn't compile, heres the error:

```
error[E0277]: the trait bound `ValueBag<'_>: From<u128>` is not satisfied
   --> /home/hipy/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:364:21
    |
364 |                       Value::from_value_bag(value)
    |                       ^^^^^^^^^^^^^^^^^^^^^ the trait `From<u128>` is not implemented for `ValueBag<'_>`
...
384 | / impl_to_value_primitive![
385 | |     usize, u8, u16, u32, u64, u128, isize, i8, i16, i32, i64, i128, f32, f64, char, bool,
386 | | ];
    | |_- in this macro invocation
    |
    = help: the following implementations were found:
              <ValueBag<'v> as From<&'a ()>>
              <ValueBag<'v> as From<&'a bool>>
              <ValueBag<'v> as From<&'a char>>
              <ValueBag<'v> as From<&'a f32>>
            and 29 others
    = note: required because of the requirements on the impl of `Into<ValueBag<'_>>` for `u128`
note: required by a bound in `value::Value::<'v>::from_value_bag`
   --> /home/hipy/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.14/src/kv/value.rs:250:12
    |
248 |     fn from_value_bag<T>(value: T) -> Self
    |        -------------- required by a bound in this
249 |     where
250 |         T: Into<ValueBag<'v>>,
    |            ^^^^^^^^^^^^^^^^^^ required by this bound in `value::Value::<'v>::from_value_bag`
    = note: this error originates in the macro `impl_to_value_primitive` (in Nightly builds, run with -Z macro-backtrace for more info)
```

It reports the same compilation error for `i128` (replacing everywhere `u128`) appears.

### Reproduces How Often

Always, can't get it to work in any of my attempts.

## Versions

**OS**: Pop!_OS (Linux)
**DB**: SQLite

**Dependencies**:
```
│   └── sea-orm v0.7.1
│       ├── sea-orm-macros v0.7.0 (proc-macro)
│       ├── sea-query v0.23.0
│       │   ├── sea-query-derive v0.2.0 (proc-macro)
│       ├── sea-strum v0.23.0
│       │   └── sea-strum_macros v0.23.0 (proc-macro)
│   └── sea-schema v0.7.1
│       ├── sea-orm v0.7.1 (*)
│       ├── sea-query v0.22.0
│       │   └── sea-query-derive v0.2.0 (proc-macro) (*)
│       ├── sea-schema-derive v0.1.0 (proc-macro)
├── sea-orm-rocket v0.5.0 (https://github.com/SeaQL/sea-orm#86e7e808)
│   └── sea-orm-rocket-codegen v0.5.0 (proc-macro) (https://github.com/SeaQL/sea-orm#86e7e808)
```

## Additional Information

Weirdly enough, the rocket examples works perfectly in my machine, just that my concrete code is for some reason not compiling, and I have no idea what's the exact cause.

**Comment by tyt2y3** [comment]
I think you have a dependency conflict on the `log` crate. Note that sea-orm does not directly depend on `log` (it's via the `tracing` crate) so it'd be hard to track down

**Comment by kulpae** [comment, marked as answer]
I could solve the same issue by deleting the Cargo.lock file

**Comment by pinkforest** [comment]
Cargo update is good for this
